scripts/bluesky-0MQ-proxy

```diff
@@ -1,5 +1,40 @@
 #!/usr/bin/env python3
+import threading
 import argparse
+import logging
+import sys
+
+from bluesky.callbacks.zmq import RemoteDispatcher
+
+def start_dispatcher(logfile):
+    """The dispatcher function
+    Parameters
+    ----------
+    logfile : string
+        string come from user command. ex --logfile=temp.log
+        logfile will be "temp.log". logfile could be empty.
+    """
+    d = RemoteDispatcher('localhost:'+str(out_port))
+    if logfile:
+        logger = logging.getLogger('logfile')
```

Comment:
The logger name is more than a label for humans; it defines a kind of "pipeline" for log messages. All the messages logged to a `'bluesky'` or `'bluesky.dispatcher'` or `'bluesky.run_engine.msg'` will "flow" to the `'bluesky'` logger because they begin with `'bluesky.*`.

At import time, we get the `'bluesky'` logger and we add a handler with a nice formatter:

https://github.com/NSLS-II/bluesky/blob/abd8552881e17d3e80313922ec040269c9780586/bluesky/log.py#L189-L190

Therefore, when messages are logged to _any_ logger than begins with `bluesky.*`, those messages will be processed by that handler handler. We only need to set up one handler at the "top" and then all of the different loggers can just use that.

So, in this case, we can do something like:

```
from bluesky import set_handler

logger = logging.getLogger('bluesky.dispatcher')
logger.setLevel('DEBUG')
if logfile:
    # Direct the logging output to our file.
    set_handler(file=logfile)
# else use the default handler, configured automatically at import time,
# which will write to the stdout
def log_write(name, doc):
    ...  # same as you have written
```

As a separate point, notice that I have suggested 'DEBUG' not 'INFO'. Since this will log _the full document_ of everything that we see, I think we should call it DEBUG logging because I can't imagine anything more verbose that we could be doing.

Reply:
My point here is when user doesn't offer --logfile option which means we just "output" to screen. Could I just use `print()` to avoid create any obj related logging which is `else` part here.